subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -56,19 +56,22 @@ If you must use AdoptOpenJDK, you should turn off auto-download. If an Eclipse T
 [[empty_directories_file_tree]]
 ==== File trees and empty directory handling
 
-When using `@SkipWhenEmpty` on an input file collection, Gradle skips the task when it determines that the input is empty.
+When using `@link:{javadocPath}/org/gradle/api/tasks/SkipWhenEmpty.html[SkipWhenEmpty]` on an input file collection, Gradle skips the task when it determines that the input is empty.
 If the input file collection consists only of file trees, Gradle ignores directories for the emptiness check.
-Though when checking for changes to the input file collection, Gradle only ignores directories when the `@IgnoreEmptyDirectories` annotation is present.
+Though when checking for changes to the input file collection, Gradle only ignores directories when the `@link:{javadocPath}/org/gradle/api/tasks/IgnoreEmptyDirectories.html[IgnoreEmptyDirectories]` annotation is present.
 
-In order to make the ignore directories behavior consistent, Gradle will now ignore directories for both the `@SkipWhenEmpty` check and for determining changes only when the `@IgnoreEmptyDirectories` annotation is present.
+To ignore directories consistently, Gradle will now ignore directories for both the `@SkipWhenEmpty` check and for determining changes only if the `@IgnoreEmptyDirectories` annotation is present.
```

Comment:
This sentence is still too complicated. Maybe break it up into multiple ones?

Reply:
Done: https://github.com/gradle/gradle/pull/18861